Improved Signup/Login UX with Loader and React-Hot-Toast

deepanshu-prajapati01 opened the pull request:
Hi team 👋

This PR addresses key UX issues in the **authentication flow** as outlined in issue #16, specifically:

---

### 🔄 What's Fixed:

#### ✅ 1. Loader Feedback on Signup/Login
- **Loading spinner** added to both Signup and Login buttons.
- **Buttons are disabled during API calls** to prevent duplicate submissions and improve feedback.

📸 *Before vs After*  
**Before**: No feedback → user clicks multiple times 
<img width="1919" height="999" alt="image" src="https://github.com/user-attachments/assets/b5b93f58-dc17-4ffd-b450-d266936326c8" />
 
**After**: Shows loading spinner and disables button
<img width="1919" height="964" alt="image" src="https://github.com/user-attachments/assets/b4605c9b-8ba0-43aa-b45b-3e53e64136c4" />

---

#### ✅ 2. Integrated `react-hot-toast` for Feedback
- Replaced all `alert()` calls with **non-blocking toast notifications**.
- Covers both success and error states (e.g., invalid input, failed API call, etc.)
- Enhances UI consistency and improves user feedback.

📸 Screenshot – Toast Example  
<img width="455" height="111" alt="image" src="https://github.com/user-attachments/assets/9a856239-3b0b-4d60-9cda-1cd0e7d7afcd" />
<img width="462" height="102" alt="image" src="https://github.com/user-attachments/assets/78cf8d36-e193-4e64-b7b6-0eb46d20d5a3" />

---

### 🏁 Summary
This PR improves the **signup/login UX** by:
- Adding visual loading feedback
- Using toast notifications for better communication

Let me know if anything else needs refinement — happy to iterate on it.

Thanks! 🚀


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added toast notification support throughout the app for improved user feedback during login, signup, email verification, and password reset.
  * Introduced "Forgot Password" and "Reset Password" flows with dedicated forms and validation.
  * Enabled resending of email verification links with cooldown timers.
  * Enhanced login and signup forms with loading indicators and disabled buttons during submission to prevent duplicate actions.
  * Simplified and improved email verification UI with success countdown and inline resend form.

* **Bug Fixes**
  * Corrected typos in route paths and error messages.
  * Fixed API endpoint URL for resending verification emails.
  * Improved error handling and user notifications using toast messages instead of alerts.

* **Chores**
  * Added "react-hot-toast" and "jwt-decode" libraries as new dependencies.
  * Added environment variable for frontend URL to support dynamic links in emails.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

deepanshu-prajapati01 committed:
Replaced alert msg with the toast

deepanshu-prajapati01 committed:
Implement loaders in signup and login forms to indicate submission progress

deepanshu-prajapati01 committed:
Disable button during network requests and update signup message to include email verification

deepanshu-prajapati01 committed:
Remove redirection to / when user sign up and better prompt email verification

deepanshu-prajapati01 commented on Frontend/src/JSX/Components/Login.jsx:
```diff
@@ -1,15 +1,17 @@
+import toast from "react-hot-toast";
+
 const Login = () => {
   const [email, setEmail] = useState("developer123@gmail.com");
   const [password, setPassword] = useState("developer@123");
```
The application is currently in development and open source, so hardcoding this is acceptable for users who want to test it out.

KrishChothani commented:
As I mention in comment,
This is not that type of work 
There is many things is incompleted.

So if you done that work than I will give you level-3 otherwise only level-1 applicable.

So if you want to work on that then let me know otherwise I merged this pr and assign level-1

KrishChothani commented:
And also there is already done the backend, 
So only need to integrate that into ui

deepanshu-prajapati01 commented:
I'm on it.

KrishChothani commented:
@deepanshu-prajapati01 
You can check www.matterassist.con

deepanshu-prajapati01 committed:
feat: include email in JWT token & add FRONTEND_URL to .env

- JWT token now carries user's email for easier frontend access during password reset or verification.
- Added FRONTEND_URL in .env for improved flexibility between dev (http://localhost:5173) and prod (https://cksedu.vercel.app) environments.

deepanshu-prajapati01 committed:
fix: corrected typos in backend routes & improved asyncHandler behavior

- Fixed typos across route files to ensure proper routing.
- Refactored asyncHandler usage to correctly propagate ApiError responses.

deepanshu-prajapati01 committed:
feat: update signup flow to support resend verification email logic

- Added resend email capability with cooldown timer
- Integrated logic into signup component for better UX

deepanshu-prajapati01 committed:
feat: added forgot password flow with email input and reset link functionality

- Implemented UI to capture user email
- Integrated backend call to send password reset email

deepanshu-prajapati01 committed:
feat: implement email verification page from scratch with robust state handling

- Displays error or success messages clearly
- Auto-redirects to /app on successful verification
- Fully maintainable and clean component structure

deepanshu-prajapati01 committed:
feat: build reset password page with full validation and update flow

- Designed page for users redirected via reset link
- Includes password + confirm password inputs with validation
- On success, password is updated via API

deepanshu-prajapati01 committed:
considered changes asked by 'github coderabbitai' bot

deepanshu-prajapati01 commented on Frontend/src/JSX/Components/ResetPassword.jsx:
```diff
@@ -0,0 +1,153 @@
+        } else {
+            setError("Missing token.");
+        }
+    }, [searchParams]);
```
done
